railties/lib/rails/generators/rails/app/templates/bin/setup

```diff
@@ -5,13 +5,17 @@ include FileUtils
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
 
+def system!(*args)
+  system(*args) || abort("\n== Command #{args} failed ==")
+end
+
 chdir APP_ROOT do
   # This script is a starting point to setup your application.
   # Add necessary setup steps to this file.
 
   puts '== Installing dependencies =='
-  system 'gem install bundler --conservative'
-  system('bundle check') or system('bundle install')
+  system! 'gem install bundler --conservative'
+  system!('bundle check') or system!('bundle install')
```

Comment:
@rsanheim I have a question :raising_hand: When `bundle check` failed, it will `abort`, does the `or system!('bundle install')` do anything?